xtensa: intel_adsp/ace30: multiple build failures

### Describe the bug

There are a lot of tests affected with the same build error : (Build failure - error: unused function 'l2_page_tables_counter_inc' [-Werror,-Wunused-function])

Problem visible only on xtensa toolchain, zephyr toolchain works

Commit that introduce problem : [xtensa: mmu/ptables: conserve memory by using COW on L2 tables](https://github.com/zephyrproject-rtos/zephyr/commit/a48345fccfdb59db58ffced4be1e0bc9dbd5d70e)

### Regression

- [x] This is a regression.

### Steps to reproduce

Just as example i have used one of arch tests : 
twister -p intel_adsp/ace30/ptl/sim -s arch.interrupt

### Relevant log output

```shell
FAILED: zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/ptables.c.obj
<short version of log>
/home/zephyr/zephyrproject/zephyr/arch/xtensa/core/ptables.c:947:27: error: unused function 'l2_page_tables_counter_inc' [-Werror,-Wunused-function]
static ALWAYS_INLINE void l2_page_tables_counter_inc(uint32_t *l2_table)
```

### Impact

Showstopper – Prevents release or major functionality; system unusable.

### Environment

Toolchain: xtensa
SKD: zephyr-sdk-0.17.4
Zephyr version: v4.3.0-748-gf1c2ce4f4026
OS: Ubuntu 22.04

### Additional Context

_No response_

--- a/arch/xtensa/core/ptables.c
+++ b/arch/xtensa/core/ptables.c
@@ -336,7 +336,7 @@ static inline uint32_t *alloc_l2_table(void)
 
 	for (idx = 0; idx < CONFIG_XTENSA_MMU_NUM_L2_TABLES; idx++) {
 		if (l2_page_tables_counter[idx] == 0) {
-			l2_page_tables_counter[idx] = 1;
+			l2_page_tables_counter_inc(l2_page_tables[idx]);
 			ret = (uint32_t *)&l2_page_tables[idx];
 			break;
 		}

